### ISSUE TITLE
Even though data and params are on GPU, computation seems to be carried out on CPU

### ISSUE THREAD

**Issue by helange23** [google/jax#5359]
JAX detects my GPUs correctly and I am able to put data and parameters on the GPU, however, when I run my code, the GPU seems to idle (1% GPU utility) while the CPU is utilized at 140%.

Is this behavior intended?

```python
from flax import linen as nn
from flax import optim
import jax.numpy as jnp
import jax

import numpy as np

def load_data():
    data = np.load('mnist.npz')
    x_train = data['x_train']
    x_train = (x_train-128.0)/128.0
    x_train = np.reshape(x_train,(60000, 28**2))
    y_train = data['y_train'][:,None]
    
    return x_train, y_train


class fnet(nn.Module):
    
    @nn.compact        
    def __call__(self, x):
        x = nn.Dense(features=25)(x)
        for i in range(10):
            x = nn.relu(nn.Dense(features=25)(x))
        return nn.Dense(features=1)(x)
    
    def initialize(self, key, x):
        return fn.init(key, x)
        
    def loss(self, x, y):
        def loss_fn(params):
            yhat = self.apply(params, x)
            loss = jnp.mean((yhat-y)**2)
            return loss
        return jax.jit(loss_fn)
            
def train_epoch(net, x_train, y_train, optimizer):

    #for i in range(len(x_train)//32):
    for i in range(300):
        
        x_tr = x_train[i*32:(i+1)*32]
        y_tr = y_train[i*32:(i+1)*32]
    
        loss = net.loss(x_tr, y_tr)
    
        v_grad = jax.value_and_grad(loss)
        val, grad = v_grad(optimizer.target)
        
        print(i, val) 
        optimizer = optimizer.apply_gradient(grad)

    return optimizer



fn = fnet()
key = jax.random.PRNGKey(1)
params = fn.initialize(key, jnp.zeros((32,28**2)))

jax.tree_map(lambda x: print(x.device_buffer.device()), params)

x_train, y_train = load_data()

rp = optim.Adam(0.0005).create(params)
x,y = jnp.array(x_train), jnp.array(y_train)

print(x.device_buffer.device(), y.device_buffer.device())
opt = train_epoch(fn, x, y, rp)
```
(I am aware that using MSE loss is stupid)

**Comment by skye** [comment]
I suggest profiling your program to see what's happening. Given that JAX detects your GPUs, all JAX operations will run on GPU, so there are probably other overheads causing the high CPU utilization.

You can use Tensorboard profiling to get a full view of what's happening on CPU + GPU (https://jax.readthedocs.io/en/latest/profiling.html), or given that you're seeing high CPU overheads, using standard Python profiling may be enough. I like to use [snakeviz](https://jiffyclub.github.io/snakeviz/) for the latter.

**Comment by ljnacht** [comment]
I tried following the installation guide for the tensorboard profiling but it couldnt find tf-nightly==2.5.0.dev20201203.
I installed tf-nightly==2.5.0.dev20201204 instead but when I try to start tensorboard, I get the following error:
"ValueError: Duplicate plugins for name projector"

Googling the error indicates that this is caused by a version mismatch. Which version of tf should I install to get the profiling to work?
